src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,180 @@
+        + "[" + PREFIX_PHONE + "] "
+        + "[" + PREFIX_EMAIL + "] "
+        + "[" + PREFIX_ADDRESS + "] "
+        + "[" + PREFIX_TAG + "] "
+        + "[" + PREFIX_ASSET + "]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_NAME + " "
+        + PREFIX_PHONE + " "
+        + PREFIX_TAG + " "
+        + PREFIX_ASSET + " ";
+
+    public static final String MESSAGE_NO_PARAM = "At least one field to copy must be provided.";
+
+    private final Index index;
+    private final boolean[] info;
+
+    /**
+     * @param index of the person in the filtered person list to copy
+     * @param info list of details to copy to clipboard
+     */
+    public CopyCommand(Index index, boolean[] info) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.info = info;
+    }
+
+    @Override
+    public String execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToCopy = lastShownList.get(index.getZeroBased());
+        String personDetails = copyToClipboard(personToCopy, info);
+
+        return String.format(Messages.MESSAGE_COPIED, personDetails);
+    }
+
+    /**
+     * Copies the details of {@code personToCopy} to clipboard
+     */
+    private static String copyToClipboard(Person personToCopy, boolean[] info) {
+        assert personToCopy != null;
+
+        StringBuilder copiedMsg = new StringBuilder();
+
+        if (info[0]) {
+            copiedMsg.append("Name: ").append(personToCopy.getName()).append("; ");
+        }
+        if (info[1]) {
+            copiedMsg.append("Phone: ").append(personToCopy.getPhone()).append("; ");
+        }
+        if (info[2]) {
+            copiedMsg.append("Email: ").append(personToCopy.getEmail()).append("; ");
+        }
+        if (info[3]) {
+            copiedMsg.append("Address: ").append(personToCopy.getAddress()).append("; ");
+        }
+        if (info[4]) {
+            copiedMsg.append("Tags: ").append(personToCopy.getTags()).append("; ");
+        }
+        if (info[5]) {
+            copiedMsg.append("Assets: ").append(personToCopy.getAssets()).append("; ");
+        }
+
+        return copiedMsg.substring(0, copiedMsg.length() - 2);
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the CopyCommand
+     * and returns an CopyCommand object for execution.
+     * @throws IllegalArgumentException if the user input does not conform the expected format
+     */
+    public static CopyCommand of(String args) throws IllegalArgumentException {
+        System.out.println(args);
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
```

Comment:
Should this be indented at 4 or 8 spaces?

Reply:
Now uses 8 spaces